tests/Aspire.Hosting.Azure.Tests/ExistingAzureResourceTests.cs

```diff
@@ -0,0 +1,194 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Utils;
+using Xunit;
+
+namespace Aspire.Hosting.Azure.Tests;
+
+public class ExistingAzureResourceTests
+{
+    [Fact]
+    public async Task AddExistingAzureServiceBusInRunMode()
+    {
+        using var builder = TestDistributedApplicationBuilder.Create();
+
+        var serviceBus = builder.AddAzureServiceBus("messaging")
+            .RunAsExisting("existing-resource-name")
+            .WithQueue("queue");
+
+        var (ManifestNode, BicepText) = await ManifestUtils.GetManifestWithBicep(serviceBus.Resource);
+
+        var expectedManifest = """
+            {
+              "type": "azure.bicep.v0",
+              "connectionString": "{messaging.outputs.serviceBusEndpoint}",
+              "path": "messaging.module.bicep",
+              "params": {
+                "principalType": "",
+                "principalId": ""
+              }
+            }
+            """;
+        Assert.Equal(expectedManifest, ManifestNode.ToString());
+
+        var expectedBicep = """
+            @description('The location for the resource(s) to be deployed.')
+            param location string = resourceGroup().location
+
+            param sku string = 'Standard'
+
+            param principalType string
+
+            param principalId string
+
+            resource messaging 'Microsoft.ServiceBus/namespaces@2024-01-01' existing = {
+              name: 'existing-resource-name'
```

Comment:
It feels like this needs to be a parameter to the bicep, and not a hard-coded name into the bicep.